ringvax/app.py

```diff
@@ -172,6 +172,52 @@ def get_commit(length: int = 15) -> Optional[str]:
         return None
 
 
+def generational_summary(
+    df: pl.DataFrame, n_sims: int, n_generations: int, alpha=0.05
+) -> alt.Chart | alt.LayerChart:
+    """Plot of number of infections by generation, summarized over simulations
+
+    Args:
+        df (pl.DataFrame): columns `simulation`, `generation`, and `num_infections`. Assumed
+          that only simulations/generation combos with >0 infections appears in the table.
+        n_sims (int): Total number of simulations (to account for missing simulation/generation
+          combos in `df`.)
+        n_generations (int): Number of generations to plot (beyond the index)
+        alpha (float, optional): Confidence interval specification. Defaults to 0.05.
+
+    Returns:
+        alt.Chart | alt.LayerChart: Plot showing median and CI across simulations, by generation
+    """
+    assert set(df.schema.names()) == {"simulation", "generation", "num_infections"}
+
+    # create a "complete" data frame, including all simulation+generation combinations
+    complete_data = (
+        pl.DataFrame({"simulation": range(n_sims)})
+        .join(pl.DataFrame({"generation": range(1, n_generations + 1)}), how="cross")
+        .join(df, on=["simulation", "generation"], how="left")
+        .with_columns(pl.col("num_infections").fill_null(value=0))
+        .sort(["simulation", "generation"])
+    )
+
+    plot_data = complete_data.group_by("generation").agg(
+        pl.col("num_infections").median().alias("median"),
+        pl.col("num_infections").quantile(alpha / 2).alias("lci"),
+        pl.col("num_infections").quantile(1.0 - alpha / 2).alias("uci"),
+    )
+
+    base = alt.Chart(plot_data)
+    cone = base.mark_area(opacity=0.3).encode(
+        x="generation:N",
+        y2="lci",
+        y=alt.Y(
+            "uci", title=f"No. of infections (median, {round(100 * (1 - alpha))}% CI)"
```

Comment:
```suggestion
            "uci", title=f"No. of infections (median, {round(100 * (1 - alpha))}% range)"
```

Not exactly happy with this rewording, but it's _not_ a CI. There's no credibility or confidence because there's no inference. If we used 50%, it would be the inter-quartile range. "Central 95% range" is a bit of a mouthful though

Reply:
Great point; implemented in https://github.com/CDCgov/cfa-ring-vax-widget/pull/84/commits/912ddc0d45cbf9c903fe770774f7e89b5c160b64